prow/sidecar/run.go

```diff
@@ -138,14 +166,23 @@ func (o Options) doUpload(spec *downwardapi.JobSpec, passed, aborted bool) error
 		result = "FAILURE"
 	}
 
+	revision := spec.Refs.BaseSHA
+	if len(spec.Refs.Pulls) > 0 {
+		revision = spec.Refs.Pulls[0].SHA
+	}
+
 	finished := struct {
-		Timestamp int64  `json:"timestamp"`
-		Passed    bool   `json:"passed"`
-		Result    string `json:"result"`
+		Timestamp  int64             `json:"timestamp"`
+		Passed     bool              `json:"passed"`
+		Result     string            `json:"result"`
+		Metadata   map[string]string `json:"metadata,omit_empty"`
+		JobVersion string            `json:"job-version"`
```

Comment:
"job-version" seems like a misleading name for the pull sha. Also note that this won't work as intended for batch jobs.

Reply:
Testgrid also looks for `repo-commit` in metadata.json which is populated in https://github.com/kubernetes/test-infra/blob/be93b6afb606bef48f52c4b8936e673d8a0b4bda/jenkins/bootstrap.py#L500-L503, I'll switch to that

(We don't really care about the commit SHA for presubmit jobs though as they are meaning less...)

Reply:
there's also some tangled logic in bootstrap...

https://github.com/kubernetes/test-infra/blob/be93b6afb606bef48f52c4b8936e673d8a0b4bda/jenkins/bootstrap.py#L539-L559

I'll try to clean this up